scripts/src/java/org/oppia/android/scripts/common/MavenDependenciesListGenerator.kt

```diff
@@ -0,0 +1,333 @@
+    }
+  }
+
+  /**
+   * Returns the set of licenses that do not have verified_link set and there original link is
+   * not set to be invalid by the developers.
+   */
+  fun getAllBrokenLicenses(
+    mavenDependenciesList: List<MavenDependency>
+  ): Set<License> {
+    // Here broken licenses are those licenses that do not have verified_link set and
+    // there original link is not set to be invalid by the developers.
+    return mavenDependenciesList.flatMap { dependency ->
+      dependency.licenseList.filter { license ->
+        license.verifiedLinkCase.equals(License.VerifiedLinkCase.VERIFIEDLINK_NOT_SET) &&
+          !license.isOriginalLinkInvalid
+      }
+    }.toSet()
+  }
+
+  /**
+   * Generates a map that maps broken licenses to the first dependency that should be updated
+   * manually in order to update all occurences of this license.
+   *
+   * @param mavenDependenciesList final list of dependencies
+   * @param brokenLicenses set of licenses that do not have verified_link set and there original
+   *    link is not set to be invalid by the developers.
+   *
+   * @return map that maps a broken license to the first dependency in maven_dependencies.textproto
+   */
+  fun findFirstDependenciesWithBrokenLicenses(
+    mavenDependenciesList: List<MavenDependency>,
+    brokenLicenses: Set<License>
+  ): Map<License, String> {
+    return brokenLicenses.associateWith { license ->
+      mavenDependenciesList.first { dependency -> license in dependency.licenseList }.artifactName
+    }
+  }
+
+  /**
+   * Returns the set of dependencies whose license list is empty or some of their license link was
+   * found to be invalid.
+   */
+  fun getDependenciesThatNeedIntervention(
+    mavenDependenciesList: List<MavenDependency>
+  ): Set<MavenDependency> {
+    // The dependencies whose license list is empty or some of their license link was found to
+    // be invalid need further intervention. In this case, the developer needs to manually fill in
+    // the license links for each of these dependencies.
+    return mavenDependenciesList.filter { dependency ->
+      dependency.licenseList.isEmpty() ||
+        dependency.licenseList.filter { license ->
+          license.verifiedLinkCase.equals(License.VerifiedLinkCase.VERIFIEDLINK_NOT_SET) &&
+            license.isOriginalLinkInvalid
+        }.isNotEmpty()
+    }.toSet()
+  }
+
+  /**
+   * Retrieve the list of maven dependencies from maven_dependencies.textproto.
+   *
+   * @param pathToPbFile path to the pb file to be parsed
+   *
+   * @return list of dependencies
+   */
+  fun retrieveMavenDependencyList(pathToPbFile: String): List<MavenDependency> {
+    return parseTextProto(
+      pathToPbFile,
+      MavenDependencyList.getDefaultInstance()
+    ).mavenDependencyList
+  }
+
+  /**
+   * Retrieve the list of [MavenListDependency] such that each of these dependencies is present
+   * in maven_install.json and is also obtained from the Bazel Query.
```

Comment:
```suggestion
   * Retrieves the list of [MavenListDependency] such that each of these dependencies is present
   * in maven_install.json and is also obtained from the Bazel query.
```

Ditto elsewhere (particularly s/Query/query).

Reply:
Done.